airflow/migrations/versions/0129_2_8_0_add_clear_number_to_dag_run.py

```diff
@@ -38,15 +37,22 @@
 
 def upgrade():
     """Apply add cleared column to dagrun"""
-    with op.batch_alter_table("dag_run") as batch_op:
-        batch_op.add_column(
-            sa.Column(
-                "clear_number",
-                sa.Integer,
-                default=0,
-                nullable=False,
+    conn = op.get_bind()
+    if conn.dialect.name == "mssql":
+        with op.batch_alter_table("dag_run") as batch_op:
+            batch_op.add_column(sa.Column("clear_number", sa.Integer, default=0))
+            batch_op.alter_column("clear_number", existing_type=sa.Integer, nullable=False)
```

Comment:
Thanks for fixing the issue but this doesn't fixed mssql.   Is there a reason why you forked the code to support this?   
alter_column can take a 'server_default' value as well.

Reply:
Discussed here: https://github.com/apache/airflow/pull/34335/files/ae9920a4be57acf30ee6a7923d3a1e4708a1c36d

Gist: _Issue was with downgrade._